dynamicio/io/postgres.py

```diff
@@ -91,34 +79,10 @@ def final_table_name(self) -> str:
         """Return schema and table name in a format of schema.table_name."""
         return f"{self.db_schema}.{self.table_name}"
 
-    def inject(self, **kwargs) -> "PostgresResource":
-        """Inject variables into stuff. Not in place."""
-        clone = deepcopy(self)
-        clone.db_user = inject(clone.db_user, **kwargs)
-        clone.db_password = inject(clone.db_password, **kwargs)
-        clone.db_host = inject(clone.db_host, **kwargs)
-        clone.db_name = inject(clone.db_name, **kwargs)
-        clone.table_name = inject(clone.table_name, **kwargs)
-        clone.sql_query = inject(clone.sql_query, **kwargs)
-        clone.db_schema = inject(clone.db_schema, **kwargs)
-        clone.test_path = inject(clone.test_path, **kwargs)
-        return clone
-
-    def check_injections(self) -> None:
-        """Check that all injections have been completed."""
-        check_injections(self.db_user)
-        check_injections(self.db_password)
-        check_injections(self.db_host)
-        check_injections(self.db_name)
-        check_injections(self.table_name)
-        check_injections(self.sql_query)
-        check_injections(self.db_schema)
-
-    def read(self) -> pd.DataFrame:
+    def _read(self) -> pd.DataFrame:
         """Handles Read operations for Postgres."""
-        self.check_injections()
         if not (bool(self.sql_query) ^ bool(self.table_name)):  # Xor
```

Comment:
judging by the `ValueError` message it seems that an `OR` could be enough here, no?